src/Symfony/Component/Serializer/Mapping/Loader/YamlFileLoader.php

```diff
@@ -87,11 +87,29 @@ public function loadClassMetadata(ClassMetadataInterface $classMetadata)
                 }
 
                 if (isset($data['serialized_name'])) {
-                    if (!\is_string($data['serialized_name']) || empty($data['serialized_name'])) {
-                        throw new MappingException(sprintf('The "serialized_name" value must be a non-empty string in "%s" for the attribute "%s" of the class "%s".', $this->file, $attribute, $classMetadata->getName()));
+                    if (!\is_string($data['serialized_name']) && !\is_array($data['serialized_name'])) {
+                        throw new MappingException(sprintf('The "serialized_name" value must be a non-empty string or an array of serialized name/groups in "%s" for the attribute "%s" of the class "%s".', $this->file, $attribute, $classMetadata->getName()));
                     }
 
-                    $attributeMetadata->setSerializedName($data['serialized_name']);
+                    if (\is_string($data['serialized_name'])) {
+                        if (empty($data['serialized_name'])) {
+                            throw new MappingException(sprintf('The "serialized_name" value must be a non-empty string in "%s" for the attribute "%s" of the class "%s".', $this->file, $attribute, $classMetadata->getName()));
+                        }
+                        $attributeMetadata->addSerializedName($data['serialized_name']);
+                    }
+
+                    if (\is_array($data['serialized_name'])) {
+                        if (empty($data['serialized_name'])) {
+                            throw new MappingException(sprintf('The "serialized_name" value must be a non-empty array of serialized name/groups in "%s" for the attribute "%s" of the class "%s".', $this->file, $attribute, $classMetadata->getName()));
+                        }
+                        foreach ($data['serialized_name'] as $serializedName => $groups) {
+                            if (!\is_string($serializedName) || empty($serializedName)) {
+                                throw new MappingException(sprintf('The key for "serialized_name" array must be a non-empty string in "%s" for the attribute "%s" of the class "%s".', $this->file, $attribute, $classMetadata->getName()));
+                            }
+
+                            $attributeMetadata->addSerializedName($serializedName, \is_array($groups) ? $groups : []);
```

Comment:
```php
(array) $groups
```
Would do same, but how it's possible that those groups are not array here?

Reply:
possible as we can accept serializedName : 'a' here as we do for group annotation for example
but I cast it as you suggest
done